crates/typst-library/src/text/mod.rs

```diff
@@ -183,9 +183,16 @@ pub struct TextElem {
         let paint: Option<Spanned<Paint>> = args.named_or_find("fill")?;
         if let Some(paint) = &paint {
             // TODO: Implement gradients on text.
-            if matches!(paint.v, Paint::Gradient(_)) {
-                bail!(error!(paint.span, "text fill must be a solid color")
-                    .with_hint("gradients on text will be supported soon"));
+            if let Paint::Gradient(gradient) = &paint.v {
+                if gradient.relative() == Smart::Custom(Relative::Self_) {
+                    bail!(
+                        error!(
+                            paint.span,
+                            "text fill must be a solid color or a gradient relative to the parent"
+                        )
+                        .with_hint("make sure to set `relative: auto` on your text fill")
```

Comment:
Unrelated to this PR, but seeing it I think `bail!` needs a named `hint: ` argument.